naga/src/valid/function.rs

```diff
@@ -1039,77 +1053,73 @@ impl super::Validator {
                         }
                     };
 
-                    // Punch through a binding array to get the underlying type
-                    let global_ty = match context.types[var.ty].inner {
-                        Ti::BindingArray { base, .. } => &context.types[base].inner,
-                        ref inner => inner,
+                    // The `image` operand must be an `Image`.
+                    let Ti::Image {
+                        class,
+                        arrayed,
+                        dim,
+                    } = context.types[image_ty].inner
+                    else {
```

Comment:
I find that `let`-`else` looks odd but maybe I just need to get used to it.

Something I wrote in a previous review:

> I'm not a fan of `if let` `else` because the thing being matched is after the pattern (reversed compared to a `match`) which disrupts the flow when reading the code. I find `let` `else` to be even worse in this regard because there is no indication that it's a condition unless you look for the `else`.
> 
> I see value in using `if let` `else` & `let` `else` when they are used for pattern matching + binding but I think there are better alternatives when pattern matching.

_from https://github.com/gfx-rs/wgpu/pull/6424#discussion_r1808533159_

This and the other usages in this PR are the pattern matching + binding case and so I think there is value in using `let`-`else` but I still find that "there is no indication that it's a condition unless you look for the `else`" disrupting the flow when reading the code.

We shouldn't block the PR on this, I'd just like to hear your thoughts on it.

Reply:
I agree when the match condition is very long, causing the wrapping to be all weird, 

```
let Some(blah) = blah else {
    continue;
```

is amazing tho